MusicApp.PlaylistService.Application/Services/Implementations/SongService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using AutoMapper;
+using MusicApp.PlaylistService.Application.DTOs;
+using MusicApp.PlaylistService.Application.Repositories;
+using MusicApp.PlaylistService.Application.Services.Interfaces;
+using MusicApp.PlaylistService.Domain.Entities;
+
+namespace MusicApp.PlaylistService.Application.Services.Implementations;
+
+public class SongService : ISongService
+{
+    private readonly ISongRepository _songRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IMapper _mapper;
+
+    public SongService(ISongRepository songRepository, IUserRepository userRepository, IMapper mapper)
+    {
+        _songRepository = songRepository;
+        _userRepository = userRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<IEnumerable<SongOutputDto>> UpdateSongs(IEnumerable<Song> songs, CancellationToken cancellationToken)
```

Comment:
i don't fully understand this logic, explain please

Reply:
The playlist service controller has a method that collects songs from the song service database (via gRPC) and calls the UpdateSongs method.

The method should synchronize the song table in the song and playlist services databases.

I decided that it would be better to delete all the rows from the song table, and then insert the songs that arrived.